diff -urN balsa-2.0.11.orig/configure.in balsa-2.0.11/configure.in
--- balsa-2.0.11.orig/configure.in	Thu May 22 22:37:21 2003
+++ balsa-2.0.11/configure.in	Thu May 22 22:37:43 2003
@@ -437,7 +437,7 @@
 set_more_warnings="$enableval", set_more_warnings=yes)
 
 if test "$GCC" = "yes" -a "$set_more_warnings" != "no"; then
-	for option in -Wsign-promo -Wsign-compare -Wpointer-arith -Wnested-externs -Wchar-subscripts -Wuninitialized -Wmissing-prototypes -Wreturn-type; do
+	for option in -Wsign-compare -Wpointer-arith -Wnested-externs -Wchar-subscripts -Wuninitialized -Wmissing-prototypes -Wreturn-type; do
                 SAVE_CFLAGS="$CFLAGS"
                 CFLAGS="$CFLAGS $option"
                 AC_MSG_CHECKING([whether gcc understands $option])
