diff -urN balsa-2.0.3.orig/configure.in balsa-2.0.3/configure.in
--- balsa-2.0.3.orig/configure.in	Wed Dec 18 21:01:37 2002
+++ balsa-2.0.3/configure.in	Wed Dec 18 22:10:00 2002
@@ -64,8 +64,8 @@
 gtk+-2.0
 gnome-vfs-2.0
 gnome-vfs-module-2.0
-libgnomeprint-2.0 >= 1.106.0
-libgnomeprintui-2.0 >= 1.106.0
+libgnomeprint-2.2 >= 2.1.5
+libgnomeprintui-2.2 >= 2.1.5
 $gnome_extras
 ])
 AC_SUBST(BALSA_CFLAGS)
diff -urN balsa-2.0.3.orig/src/print.c balsa-2.0.3/src/print.c
--- balsa-2.0.3.orig/src/print.c	Wed Dec 18 21:01:37 2002
+++ balsa-2.0.3/src/print.c	Wed Dec 18 22:20:13 2002
@@ -30,10 +30,10 @@
 #include <ctype.h>
 #include <libgnomeprint/gnome-print.h>
 #include <libgnomeprint/gnome-font.h>
-#include <libgnomeprint/gnome-print-master.h>
+#include <libgnomeprint/gnome-print-job.h>
 #include <libgnomeprint/gnome-print-paper.h>
 #include <libgnomeprintui/gnome-print-dialog.h>
-#include <libgnomeprintui/gnome-print-master-preview.h>
+#include <libgnomeprintui/gnome-print-job-preview.h>
 #include <libbalsa.h>
 #ifdef HAVE_PCRE
 #  include <pcreposix.h>
@@ -990,18 +990,18 @@
 }
 
 static PrintInfo *
-print_info_new(LibBalsaMessage * msg, GnomePrintMaster * master,
+print_info_new(LibBalsaMessage * msg, GnomePrintJob * master,
                CommonInfo * ci)
 {
     GnomePrintConfig* config;
     PrintInfo *pi = g_new0(PrintInfo, 1);
 
-    config = gnome_print_master_get_config(master);
-    gnome_print_master_get_page_size_from_config(config, &pi->page_width,
+    config = gnome_print_job_get_config(master);
+    gnome_print_job_get_page_size_from_config(config, &pi->page_width,
                                                  &pi->page_height);
     gnome_print_config_unref(config);
 
-    pi->pc = gnome_print_master_get_context(master);
+    pi->pc = gnome_print_job_get_context(master);
     pi->margin_top = 0.75 * 72;
     pi->margin_bottom = 0.75 * 72;
     pi->margin_left = 0.75 * 72;
@@ -1162,7 +1162,7 @@
  * creates the print dialog, and adds a page for fonts
  */
 static GtkWidget *
-print_dialog(GnomePrintMaster * master, CommonInfo * ci)
+print_dialog(GnomePrintJob * master, CommonInfo * ci)
 {
     GtkWidget  *dialog;
     GtkWidget  *frame;
@@ -1173,8 +1173,8 @@
     GtkWidget  *chkbut;
     GList      *childList;
 
-    dialog = gnome_print_dialog_new_from_master(master, _("Print message"),
-				                GNOME_PRINT_DIALOG_COPIES);
+    dialog = gnome_print_dialog_new(master, _("Print message"),
+				    GNOME_PRINT_DIALOG_COPIES);
     gtk_window_set_wmclass(GTK_WINDOW(dialog), "print", "Balsa");
     dlgVbox = GTK_DIALOG(dialog)->vbox;
     childList = gtk_container_get_children(GTK_CONTAINER(dlgVbox));
@@ -1290,7 +1290,7 @@
 message_print(LibBalsaMessage * msg)
 {
     CommonInfo ci;
-    GnomePrintMaster *master;
+    GnomePrintJob *master;
     GnomePrintConfig* config;
     PrintInfo *pi;
     gboolean preview;
@@ -1299,14 +1299,14 @@
 
     common_info_setup(&ci);
 
-    master = gnome_print_master_new();
+    master = gnome_print_job_new(NULL);
 
     /* FIXME: this sets the paper size in the GnomePrintConfig. We can
      * change it in the Paper page of the GnomePrintDialog, and retrieve
      * it from the GnomePrintConfig. However, it doesn't get set as the
      * initial value in the Paper page. Is there some Gnome-2-wide
      * repository for data like this? */
-    config = gnome_print_master_get_config(master);
+    config = gnome_print_job_get_config(master);
     gnome_print_config_set(config, GNOME_PRINT_KEY_PAPER_SIZE, 
                            balsa_app.paper_size);
     
@@ -1334,16 +1334,16 @@
 
     /* do the Real Job */
     print_message(pi);
-    gnome_print_master_close(master);
+    gnome_print_job_close(master);
     if (preview) {
 	GtkWidget *preview_widget =
-	    gnome_print_master_preview_new(master,
+	    gnome_print_job_preview_new(master,
 		 			   _("Balsa: message print preview"));
         gtk_window_set_wmclass(GTK_WINDOW(preview_widget), "print-preview",
                                "Balsa");
 	gtk_widget_show(preview_widget);
     } else
-	gnome_print_master_print(master);
+	gnome_print_job_print(master);
 
     print_info_destroy(pi);
     common_info_cleanup(&ci);
