--- balsa-2.0.17/configure.in.orig	Sun Mar 28 23:41:56 2004
+++ balsa-2.0.17/configure.in	Fri May  7 13:26:55 2004
@@ -364,11 +364,12 @@
            [if test "$with_iconv" != yes -a "$with_iconv" != no ; then
                    CPPFLAGS="-I${with_iconv}/include $CPPFLAGS"
                    LDFLAGS="-L${with_iconv}/lib $LDFLAGS"
-            fi])
- AC_CHECK_LIB(iconv, iconv_open, , [
+            fi
+	    if test "$with_iconv" != no ; then
+	           AC_CHECK_LIB(iconv, iconv_open)
+	    fi])
  AC_CHECK_FUNC(iconv_open, , [
  	AC_MSG_ERROR([*** You need iconv for balsa.])
-  ])
 ])
 
 dnl ##########################################################################
