diff -urN balsa-2.0.3.orig/configure.in balsa-2.0.3/configure.in
--- balsa-2.0.3.orig/configure.in	Wed Dec 18 21:01:37 2002
+++ balsa-2.0.3/configure.in	Wed Dec 18 21:27:06 2002
@@ -88,9 +88,9 @@
 AC_CONFIG_SUBDIRS(libmutt)
 
 
-AC_CHECK_LIB(pspell, new_pspell_manager,,[
+AC_CHECK_LIB(pspell, new_aspell_speller,,[
 	AC_MSG_ERROR([*** libpspell(-devel) not found.  Please check http://www.balsa.net for information on the required files.])
-], -lstdc++ -lpspell-modules)
+], -lstdc++)
 
 dnl Reset
 # LDFLAGS="$balsa_save_LDFLAGS"
