--- configure.in~	Tue Nov 13 00:08:21 2001
+++ configure.in	Tue Nov 13 00:09:00 2001
@@ -247,7 +247,7 @@
 AC_MSG_CHECKING(whether to use LDAP)
 if test x"$use_ldap" = xyes ; then
 	AC_MSG_RESULT([yes])
-	AC_CHECK_LIB(ldap, ldap_search, AC_DEFINE(ENABLE_LDAP) LIBS="-lldap -llber $LIBS", AC_MSG_ERROR([*** You enabled LDAP but ldap library is not found.]), -llber)
+	AC_CHECK_LIB(ldap, ldap_search, AC_DEFINE(ENABLE_LDAP) LIBS="-lldap -llber -lstdc++ $LIBS", AC_MSG_ERROR([*** You enabled LDAP but ldap library is not found.]), -llber -lstdc++)
 else
 	AC_MSG_RESULT([no])
 fi
--- acinclude.m4~	Tue Nov 13 00:24:14 2001
+++ acinclude.m4	Tue Nov 13 00:24:35 2001
@@ -19,3 +19,4 @@
    AC_DEFINE(HAVE_SIGSET_H)
 fi
 ])
+AC_DEFUN(GOB_CHECK,:)
