--- balsa-1.1.7.new/libinit_balsa/Makefile.am~	Tue Oct  3 15:42:38 2000
+++ balsa-1.1.7.new/libinit_balsa/Makefile.am	Sun Sep  9 21:24:57 2001
@@ -1,14 +1,6 @@
 noinst_LIBRARIES = libinit_balsa.a
 
 libinit_balsa_a_SOURCES = 		\
-	$(BUILT_SOURCES)		\
-	$(GOB_SOURCES)			\
-	helper.c			\
-	helper.h			\
-	init_balsa.c			\
-	init_balsa.h
-
-BUILT_SOURCES =				\
 	balsa-initdruid.c		\
 	balsa-druid-page.c		\
 	balsa-druid-page-error.c	\
@@ -22,16 +14,18 @@
 	balsa-druid-page-welcome.h	\
 	balsa-druid-page-user.h		\
 	balsa-druid-page-directory.h	\
-	balsa-druid-page-finish.h	
-
-GOB_SOURCES =				\
+	balsa-druid-page-finish.h	\
 	balsa-initdruid.gob		\
 	balsa-druid-page.gob		\
 	balsa-druid-page-error.gob	\
 	balsa-druid-page-welcome.gob	\
 	balsa-druid-page-user.gob	\
 	balsa-druid-page-directory.gob	\
-	balsa-druid-page-finish.gob
+	balsa-druid-page-finish.gob	\
+	helper.c			\
+	helper.h			\
+	init_balsa.c			\
+	init_balsa.h
 
 INCLUDES=-I. -I$(top_srcdir) \
 	-I$(top_builddir)/libmutt \
--- balsa-1.1.7.new/src/Makefile.am~	Thu Jul 12 09:08:46 2001
+++ balsa-1.1.7.new/src/Makefile.am	Sun Sep  9 21:35:34 2001
@@ -1,6 +1,6 @@
 bin_PROGRAMS = balsa
 
-balsa_BASE_SOURCES =	\
+balsa_SOURCES = 		\
 	address-book.c		\
 	address-book.h		\
 	address-book-config.c	\
@@ -52,14 +52,6 @@
 	spell-check.c		\
 	spell-check.h		
 
-balsa_IDL_SOURCES = \
-	balsa-common.c		\
-	balsa-skels.c		\
-	balsa-stubs.c		
-balsa_IDL_SOURCES = 
-
-balsa_SOURCES = $(balsa_BASE_SOURCES) $(balsa_IDL_SOURCES)
-
 INCLUDES = \
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" \
 	-I$(top_srcdir) \
@@ -67,8 +59,6 @@
 	-I$(top_srcdir)/libmutt \
 	 $(GNOME_INCLUDEDIR)
 
-CFLAGS += $(GTKHTML_CFLAGS) $(PIXBUF_FLAGS)
-
 balsa_LDADD = \
 	$(top_builddir)/libbalsa/libbalsa.a	\
 	$(top_builddir)/libmutt/imap/libimap.a	\
@@ -76,7 +66,7 @@
 	$(top_builddir)/libinit_balsa/libinit_balsa.a \
 	-lpspell \
 	-lltdl \
-	@LIBESD_LIB@ \
+	$(LIBESD_LIB) \
 	$(GNOME_LIBDIR) \
 	$(GNOMEUI_LIBS) \
 	$(INTLLIBS) \
@@ -85,27 +75,9 @@
 	$(GTKHTML_LIBS)	\
 	$(PIXBUF_LIBS)
 
-balsa_DEPENDENCIES = \
-	$(top_builddir)/libbalsa/libbalsa.a	\
-	$(top_builddir)/libmutt/libmutt.a	\
-        $(top_builddir)/libmutt/imap/libimap.a	\
-	$(top_builddir)/libmutt/libmutt.a	\
-	$(top_builddir)/libinit_balsa/libinit_balsa.a
+balsa_CFLAGS = $(GTKHTML_CFLAGS) $(PIXBUF_FLAGS)
 
 dist-hook:
 	mkdir $(distdir)/pixmaps
 	cp $(srcdir)/pixmaps/*.xpm $(distdir)/pixmaps
 
-idldir = $(datadir)/idl
-
-idl_DATA = #../idl/balsa.idl
-
-CORBA_SRCLIST = balsa-common.c balsa-skels.c balsa-stubs.c \
-                balsa.h
-
-
-BALSA_ORB_SRCLIST = orbit-glue.c 
-
-
-$(CORBA_SRCLIST): $(idl_DATA)
-		$(ORBIT_IDL) $(idl_DATA)
--- balsa-1.1.7.new/libbalsa/Makefile.am~	Thu Jul 12 09:08:46 2001
+++ balsa-1.1.7.new/libbalsa/Makefile.am	Sun Sep  9 21:38:13 2001
@@ -62,8 +62,5 @@
 	server.h		\
 	threads.h
 
-
-libbalsa_a_LIBADD = @LIBOBJS@
-
 INCLUDES=-I. -I${top_srcdir} -I${top_builddir}/libmutt \
 	$(GNOME_INCLUDEDIR)
--- balsa-1.1.7/images/Makefile.am~	Wed Mar 29 05:14:44 2000
+++ balsa-1.1.7/images/Makefile.am	Sun Sep  9 21:54:08 2001
@@ -2,7 +2,7 @@
 
 EXTRA_DIST =	$(balsa_BITMAPS)
 
-balsabitmapsdir = @gnomedatadir@/pixmaps/balsa
+balsabitmapsdir = $(gnomedatadir)/pixmaps/balsa
 
 balsabitmaps_DATA = $(balsa_BITMAPS)
 
--- balsa-1.1.7/sounds/Makefile.am~	Thu Apr 13 19:38:24 2000
+++ balsa-1.1.7/sounds/Makefile.am	Sun Sep  9 21:56:54 2001
@@ -3,8 +3,8 @@
 
 EXTRA_DIST = $(balsa_SOUNDS) $(balsa_SOUNDLISTS)
 
-balsasoundsdir = @gnomedatadir@/sounds/balsa
+balsasoundsdir = $(gnomedatadir)/sounds/balsa
 balsasounds_DATA = $(balsa_SOUNDS)
 
-soundlistdir = @gnomeconfdir@/sound/events
+soundlistdir = $(gnomeconfdir)/sound/events
 soundlist_DATA = $(balsa_SOUNDLISTS)
